From: Ed Swierk <eswierk@aristanetworks.com>
Date: Tue, 7 Apr 2009 00:49:12 +0000 (-0700)
Subject: forcedeth: Fix resume from hibernation regression.
X-Git-Tag: v2.6.30-rc1~137^2~1
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=35a7433c789ba6df6d96b70fa745ae9e6cac0038

forcedeth: Fix resume from hibernation regression.

Reset phy state on resume, fixing a regression caused by powering down
the phy on hibernate.

Signed-off-by: Ed Swierk <eswierk@aristanetworks.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[cebbert@redhat.com: backport to 2.6.29]
---

diff --git a/drivers/net/forcedeth.c b/drivers/net/forcedeth.c
index a858c6f..663a7b3 100644
--- a/drivers/net/forcedeth.c
+++ b/drivers/net/forcedeth.c
@@ -6117,6 +6117,9 @@ static int nv_resume(struct pci_dev *pdev)
 	for (i = 0;i <= np->register_size/sizeof(u32); i++)
 		writel(np->saved_config_space[i], base+i*sizeof(u32));
 
+	/* restore phy state, including autoneg */
+	phy_init(dev);
+
 	netif_device_attach(dev);
 	if (netif_running(dev)) {
 		rc = nv_open(dev);
